<!DOCTYPE html>
<html>
<head><title>R: TCGAbiolinks: An R/Bioconductor package for integrative analysis
with GDC data</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">
<h1> TCGAbiolinks: An R/Bioconductor package for integrative analysis
with GDC data
<img class="toplogo" src="../../../doc/html/Rlogo.svg" alt="[R logo]" />
</h1>
<hr/>
<div style="text-align: center;">
<a href="../../../doc/html/packages.html"><img class="arrow" src="../../../doc/html/left.jpg" alt="[Up]" /></a>
<a href="../../../doc/html/index.html"><img class="arrow" src="../../../doc/html/up.jpg" alt="[Top]" /></a>
</div><h2>Documentation for package &lsquo;TCGAbiolinks&rsquo; version 2.37.1</h2>

<ul><li><a href="../DESCRIPTION">DESCRIPTION file</a>.</li>
<li><a href="../NEWS">Package NEWS</a>.</li>
</ul>

<h2>Help Pages</h2>


<table style="width: 100%;">
<tr><td style="width: 25%;"><a href="colDataPrepare.html">colDataPrepare</a></td>
<td>Create samples information matrix for GDC samples</td></tr>
<tr><td style="width: 25%;"><a href="dmc.non.parametric.html">dmc.non.parametric</a></td>
<td>Perform non-parametrix wilcoxon test</td></tr>
<tr><td style="width: 25%;"><a href="gaiaCNVplot.html">gaiaCNVplot</a></td>
<td>Creates a plot for GAIA output (all significant aberrant regions.)</td></tr>
<tr><td style="width: 25%;"><a href="GDCdownload.html">GDCdownload</a></td>
<td>Download GDC data</td></tr>
<tr><td style="width: 25%;"><a href="GDCprepare.html">GDCprepare</a></td>
<td>Prepare GDC data</td></tr>
<tr><td style="width: 25%;"><a href="GDCprepare_clinic.html">GDCprepare_clinic</a></td>
<td>Parsing clinical xml files</td></tr>
<tr><td style="width: 25%;"><a href="GDCquery.html">GDCquery</a></td>
<td>Query GDC data</td></tr>
<tr><td style="width: 25%;"><a href="GDCquery_ATAC_seq.html">GDCquery_ATAC_seq</a></td>
<td>Retrieve open access ATAC-seq files from GDC server</td></tr>
<tr><td style="width: 25%;"><a href="GDCquery_clinic.html">GDCquery_clinic</a></td>
<td>Get GDC clinical data</td></tr>
<tr><td style="width: 25%;"><a href="get.GRCh.bioMart.html">get.GRCh.bioMart</a></td>
<td>Get hg19 gene annotation or hg38 (gencode v36)</td></tr>
<tr><td style="width: 25%;"><a href="getAdjacencyBiogrid.html">getAdjacencyBiogrid</a></td>
<td>Get a matrix of interactions of genes from biogrid</td></tr>
<tr><td style="width: 25%;"><a href="getDataCategorySummary.html">getDataCategorySummary</a></td>
<td>Create a Summary table for each sample in a project saying if it contains or not files for a certain data category</td></tr>
<tr><td style="width: 25%;"><a href="getGDCInfo.html">getGDCInfo</a></td>
<td>Check GDC server status</td></tr>
<tr><td style="width: 25%;"><a href="getGDCprojects.html">getGDCprojects</a></td>
<td>Retrieve all GDC projects</td></tr>
<tr><td style="width: 25%;"><a href="getGistic.html">getGistic</a></td>
<td>Download GISTIC data from firehose</td></tr>
<tr><td style="width: 25%;"><a href="getLinkedOmicsData.html">getLinkedOmicsData</a></td>
<td>Retrieve linkedOmics data</td></tr>
<tr><td style="width: 25%;"><a href="getManifest.html">getManifest</a></td>
<td>Get a Manifest from GDCquery output that can be used with GDC-client</td></tr>
<tr><td style="width: 25%;"><a href="getMC3MAF.html">getMC3MAF</a></td>
<td>Retrieve open access mc3 MAF file from GDC server</td></tr>
<tr><td style="width: 25%;"><a href="getNbCases.html">getNbCases</a></td>
<td>Get Number of cases in GDC for a project</td></tr>
<tr><td style="width: 25%;"><a href="getNbFiles.html">getNbFiles</a></td>
<td>Get Number of files in GDC for a project</td></tr>
<tr><td style="width: 25%;"><a href="getProjectSummary.html">getProjectSummary</a></td>
<td>Get Project Summary from GDC</td></tr>
<tr><td style="width: 25%;"><a href="getResults.html">getResults</a></td>
<td>Get the results table from query</td></tr>
<tr><td style="width: 25%;"><a href="getSampleFilesSummary.html">getSampleFilesSummary</a></td>
<td>Retrieve summary of files per sample in a project</td></tr>
<tr><td style="width: 25%;"><a href="getTSS.html">getTSS</a></td>
<td>getTSS to fetch GENCODE gene annotation (transcripts level) from Bioconductor package biomaRt If upstream and downstream are specified in TSS list, promoter regions of GENCODE gene will be generated.</td></tr>
<tr><td style="width: 25%;"><a href="get_IDs.html">get_IDs</a></td>
<td>Extract information from TCGA barcodes.</td></tr>
<tr><td style="width: 25%;"><a href="gliomaClassifier.html">gliomaClassifier</a></td>
<td>Gliomar classifier</td></tr>
<tr><td style="width: 25%;"><a href="isServeOK.html">isServeOK</a></td>
<td>Check GDC server status is OK</td></tr>
<tr><td style="width: 25%;"><a href="matchedMetExp.html">matchedMetExp</a></td>
<td>Get GDC primary tumors samples with both DNA methylation (HM450K) and Gene expression data</td></tr>
<tr><td style="width: 25%;"><a href="PanCancerAtlas_subtypes.html">PanCancerAtlas_subtypes</a></td>
<td>Retrieve table with TCGA molecular subtypes</td></tr>
<tr><td style="width: 25%;"><a href="splitAPICall.html">splitAPICall</a></td>
<td>internal function to break a huge API call into smaller ones so it repects the max character limit of a string</td></tr>
<tr><td style="width: 25%;"><a href="TabSubtypesCol_merged.html">TabSubtypesCol_merged</a></td>
<td>TCGA samples with their Pam50 subtypes</td></tr>
<tr><td style="width: 25%;"><a href="TCGAanalyze_analyseGRN.html">TCGAanalyze_analyseGRN</a></td>
<td>Generate network</td></tr>
<tr><td style="width: 25%;"><a href="TCGAanalyze_Clustering.html">TCGAanalyze_Clustering</a></td>
<td>Hierarchical cluster analysis</td></tr>
<tr><td style="width: 25%;"><a href="TCGAanalyze_DEA.html">TCGAanalyze_DEA</a></td>
<td>Differential expression analysis (DEA) using edgeR or limma package.</td></tr>
<tr><td style="width: 25%;"><a href="TCGAanalyze_DEA_Affy.html">TCGAanalyze_DEA_Affy</a></td>
<td>Differentially expression analysis (DEA) using limma package.</td></tr>
<tr><td style="width: 25%;"><a href="TCGAanalyze_DMC.html">TCGAanalyze_DMC</a></td>
<td>Differentially methylated regions Analysis</td></tr>
<tr><td style="width: 25%;"><a href="TCGAanalyze_EA.html">TCGAanalyze_EA</a></td>
<td>Enrichment analysis of a gene-set with GO [BP,MF,CC] and pathways.</td></tr>
<tr><td style="width: 25%;"><a href="TCGAanalyze_EAcomplete.html">TCGAanalyze_EAcomplete</a></td>
<td>Enrichment analysis for Gene Ontology (GO) [BP,MF,CC] and Pathways</td></tr>
<tr><td style="width: 25%;"><a href="TCGAanalyze_Filtering.html">TCGAanalyze_Filtering</a></td>
<td>Filtering mRNA transcripts and miRNA selecting a threshold.</td></tr>
<tr><td style="width: 25%;"><a href="TCGAanalyze_LevelTab.html">TCGAanalyze_LevelTab</a></td>
<td>Adding information related to DEGs genes from DEA as mean values in two conditions.</td></tr>
<tr><td style="width: 25%;"><a href="TCGAanalyze_networkInference.html">TCGAanalyze_networkInference</a></td>
<td>infer gene regulatory networks</td></tr>
<tr><td style="width: 25%;"><a href="TCGAanalyze_Normalization.html">TCGAanalyze_Normalization</a></td>
<td>normalization mRNA transcripts and miRNA using EDASeq package.</td></tr>
<tr><td style="width: 25%;"><a href="TCGAanalyze_Pathview.html">TCGAanalyze_Pathview</a></td>
<td>Generate pathview graph</td></tr>
<tr><td style="width: 25%;"><a href="TCGAanalyze_Preprocessing.html">TCGAanalyze_Preprocessing</a></td>
<td>Array Array Intensity correlation (AAIC) and correlation boxplot to define outlier</td></tr>
<tr><td style="width: 25%;"><a href="TCGAanalyze_Stemness.html">TCGAanalyze_Stemness</a></td>
<td>Generate Stemness Score based on RNASeq (mRNAsi stemness index) Malta et al., Cell, 2018</td></tr>
<tr><td style="width: 25%;"><a href="TCGAanalyze_survival.html">TCGAanalyze_survival</a></td>
<td>Creates survival analysis</td></tr>
<tr><td style="width: 25%;"><a href="TCGAanalyze_SurvivalKM.html">TCGAanalyze_SurvivalKM</a></td>
<td>survival analysis (SA) univariate with Kaplan-Meier (KM) method.</td></tr>
<tr><td style="width: 25%;"><a href="TCGAbatch_Correction.html">TCGAbatch_Correction</a></td>
<td>Batch correction using ComBat and Voom transformation using limma package.</td></tr>
<tr><td style="width: 25%;"><a href="TCGAprepare_Affy.html">TCGAprepare_Affy</a></td>
<td>Prepare CEL files into an AffyBatch.</td></tr>
<tr><td style="width: 25%;"><a href="TCGAquery_MatchedCoupledSampleTypes.html">TCGAquery_MatchedCoupledSampleTypes</a></td>
<td>Retrieve multiple tissue types from the same patients.</td></tr>
<tr><td style="width: 25%;"><a href="TCGAquery_recount2.html">TCGAquery_recount2</a></td>
<td>Query gene counts of TCGA and GTEx data from the Recount2 project</td></tr>
<tr><td style="width: 25%;"><a href="TCGAquery_SampleTypes.html">TCGAquery_SampleTypes</a></td>
<td>Retrieve multiple tissue types not from the same patients.</td></tr>
<tr><td style="width: 25%;"><a href="TCGAquery_subtype.html">TCGAquery_subtype</a></td>
<td>Retrieve molecular subtypes for a given tumor</td></tr>
<tr><td style="width: 25%;"><a href="TCGAtumor_purity.html">TCGAtumor_purity</a></td>
<td>Filters TCGA barcodes according to purity parameters</td></tr>
<tr><td style="width: 25%;"><a href="TCGAvisualize_BarPlot.html">TCGAvisualize_BarPlot</a></td>
<td>Barplot of subtypes and clinical info in groups of gene expression clustered.</td></tr>
<tr><td style="width: 25%;"><a href="TCGAvisualize_EAbarplot.html">TCGAvisualize_EAbarplot</a></td>
<td>barPlot for a complete Enrichment Analysis</td></tr>
<tr><td style="width: 25%;"><a href="TCGAvisualize_Heatmap.html">TCGAvisualize_Heatmap</a></td>
<td>Heatmap with more sensible behavior using heatmap.plus</td></tr>
<tr><td style="width: 25%;"><a href="TCGAvisualize_meanMethylation.html">TCGAvisualize_meanMethylation</a></td>
<td>Mean methylation boxplot</td></tr>
<tr><td style="width: 25%;"><a href="TCGAvisualize_oncoprint.html">TCGAvisualize_oncoprint</a></td>
<td>Creating a oncoprint</td></tr>
<tr><td style="width: 25%;"><a href="TCGAvisualize_PCA.html">TCGAvisualize_PCA</a></td>
<td>Principal components analysis (PCA) plot</td></tr>
<tr><td style="width: 25%;"><a href="TCGAvisualize_starburst.html">TCGAvisualize_starburst</a></td>
<td>Create starburst plot</td></tr>
<tr><td style="width: 25%;"><a href="TCGAvisualize_SurvivalCoxNET.html">TCGAvisualize_SurvivalCoxNET</a></td>
<td>Survival analysis with univariate Cox regression package (dnet)</td></tr>
<tr><td style="width: 25%;"><a href="TCGAVisualize_volcano.html">TCGAVisualize_volcano</a></td>
<td>Creates a volcano plot for DNA methylation or gene expression</td></tr>
<tr><td style="width: 25%;"><a href="TCGA_MolecularSubtype.html">TCGA_MolecularSubtype</a></td>
<td>Retrieve molecular subtypes for given TCGA barcodes</td></tr>
<tr><td style="width: 25%;"><a href="Tumor.purity.html">Tumor.purity</a></td>
<td>TCGA samples with their Tumor Purity measures</td></tr>
<tr><td style="width: 25%;"><a href="UseRaw_afterFilter.html">UseRaw_afterFilter</a></td>
<td>Use raw count from the DataPrep object which genes are removed by normalization and filtering steps.</td></tr>
</table>
</div></body></html>
